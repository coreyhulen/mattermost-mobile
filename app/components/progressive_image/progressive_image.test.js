// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {Image} from 'react-native';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import ProgressiveImage from './progressive_image';

describe('ProgressiveImage', () => {
    const baseProps = {
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <ProgressiveImage {...baseProps}/>
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should parse defaultSource uri', () => {
        const uri = 'HTTP://test-uri/ABC123';
        const expectedUri = 'http://test-uri/ABC123';

        const props = {
            ...baseProps,
            defaultSource: {uri},
        };
        const wrapper = shallow(
            <ProgressiveImage {...props}/>
        );

        const image = wrapper.find(Image);
        expect(image.props().source.uri).toBe(expectedUri);
    });
});