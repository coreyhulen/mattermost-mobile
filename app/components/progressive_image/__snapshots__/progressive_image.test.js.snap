// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressiveImage should match snapshot 1`] = `
<View
  style={Object {}}
>
  <Image
    source={Object {}}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;
